app.js

```diff
@@ -17,14 +18,22 @@ const server = new ApolloServer({
 
 app.use(cors());
 server.applyMiddleware({ app, bodyParserConfig: bodyParser.json() });
-const url = `mongodb+srv://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@cluster0-yt8sz.mongodb.net/${process.env.MONGO_DB}`;
-mongoose.connect(url, { useNewUrlParser: true })
+
+const url = env('DATABASE_URL');
```

Comment:
Will need to add this variable to circle CI for the tests to pass

Reply:
Yeah, its needed.

Reply:
@Allan690, I double checked the information regarding the `DATABASE_URL` and I realise that the test fails even when the config var is provided. And also, there's an error regarding the `port already in use`.

So I took the liberty of abstracting the server listener, and database connection to a separate file, that way we don't have to listen to a port every-time we run the test.

Reply:
Thinking about, are there going to be integration tests for model operation, or a simple mock will do?

Reply:
> @Allan690, I double checked the information regarding the `DATABASE_URL` and I realise that the test fails on my local machine even when the config var is provided. And also, there's an error `port already in use` as a result of an existing running server.
> 
> So I took the liberty of abstracting the server listener, and database connection to a separate file, that way we don't have to listen to a port every-time we run the test.

I like the implementation. Nice work there.

Reply:
> Thinking about, are there going to be integration tests for model operation, or a simple mock will do?

A simple mock could do for now.

Reply:
Alright, cool.